lib/auth/saml.go

```diff
@@ -111,7 +111,18 @@ func (a *Server) CreateSAMLAuthRequest(req services.SAMLAuthRequest) (*services.
 	}
 
 	req.ID = attr.Value
-	req.RedirectURL, err = provider.BuildAuthURLFromDocument("", doc)
+
+	// Workaround for Ping: Ping expects `SigAlg` and `Signature` query
+	// parameters when "Enforce Signed Authn Request" is enabled, but gosaml2
+	// only provides these parameters when binding == BindingHttpRedirect.
+	// Luckily, BuildAuthURLRedirect sets this and is otherwise identical to
+	// the standard BuildAuthURLFromDocument.
+	if connector.GetProvider() == teleport.Ping {
```

Comment:
Is `provider` a standard field we support in SAML connector resource spec? Should probably mention it in the docs, I couldn't find any mentions of it currently.

Reply:
It is at least mentioned in the docs regarding ADFS: https://goteleport.com/docs/enterprise/sso/ssh-adfs/#create-teleport-roles

Reply:
@timothyb89 Yes, please create a docs PR after this is merged and add @inertial-frame as reviewer. Let's document some of these workarounds we have in-place.

Reply:
I've just posted a docs PR here: https://github.com/gravitational/teleport/pull/7339